### ISSUE TITLE
truncate_words method getting hung for specific text input in Rails 4.2

### ISSUE THREAD

**Issue by jebingeosil** [rails/rails#25127]
### Steps to reproduce

sample_text =  `"<p><span>Returns a checkbox tag tailored for accessing a specified attribute (identified by </span><tt>method</tt><span>) on an object assigned to the template (identified by </span><tt>object</tt><span>). This object must be an instance object (@object) and not a local object. It&rsquo;s intended that </span><tt>method</tt><span> returns an integer and if that integer is above zero, then the checkbox is checked. Additional options on the input tag can be passed as a hash with </span><tt>options</tt><span>. The </span><tt>checked_value</tt><span>defaults to 1 while the default </span><tt>unchecked_value</tt><span> is set to 0 which is convenient for boolean values.</span></p>"`

sample_text.truncate_words(n), n > 90, eg: sample_text.truncate_words(200)

### Expected behavior
 sample_text.truncate_words(200) should return the entire string

### Actual behavior
It just hangs the server without any output

### System configuration
**Rails version**: 4.2

**Ruby version**:  2.3.1

**Comment by sivagollapalli** [comment]
@jebingeosil I just checked and it worked for me as expected. I am using `rails 4.2.5` and `ruby-2.3.1p112`. Let me know if you are using a different version.

**Comment by vipulnsward** [comment]
@jebingeosil can you create an executable script with the issue reproduction? https://github.com/rails/rails/blob/master/guides/bug_report_templates/generic_gem.rb

**Comment by jebingeosil** [comment]
@sivagollapalli  Thanks for the quick response, I am using rails 4.2.0  and ruby ruby 2.3.1p112

**Comment by rafaelfranca** [comment]
@jebingeosil could you upgrade to 4.2.6?

**Comment by vipulnsward** [comment]
On 4.2.6, it seems to work fine for me:
```ruby
irb(main):013:0> sample_text.truncate_words(200)
=> "<p><span>Returns a checkbox tag tailored for accessing a specified attribute (identified by </span><tt>method</tt><span>) on an object assigned to the template (identified by </span><tt>object</tt><span>). This object must be an instance object (@object) and not a local object. It&rsquo;s intended that </span><tt>method</tt><span> returns an integer and if that integer is above zero, then the checkbox is checked. Additional options on the input tag can be passed as a hash with </span><tt>options</tt><span>. The </span><tt>checked_value</tt><span>defaults to 1 while the default </span><tt>unchecked_value</tt><span> is set to 0 which is convenient for boolean values.</span></p>"
irb(main):014:0> sample_text.truncate_words(100)
=> "<p><span>Returns a checkbox tag tailored for accessing a specified attribute (identified by </span><tt>method</tt><span>) on an object assigned to the template (identified by </span><tt>object</tt><span>). This object must be an instance object (@object) and not a local object. It&rsquo;s intended that </span><tt>method</tt><span> returns an integer and if that integer is above zero, then the checkbox is checked. Additional options on the input tag can be passed as a hash with </span><tt>options</tt><span>. The </span><tt>checked_value</tt><span>defaults to 1 while the default </span><tt>unchecked_value</tt><span> is set to 0 which is convenient for boolean values.</span></p>"
irb(main):016:0> sample_text.truncate_words(180)
=> "<p><span>Returns a checkbox tag tailored for accessing a specified attribute (identified by </span><tt>method</tt><span>) on an object assigned to the template (identified by </span><tt>object</tt><span>). This object must be an instance object (@object) and not a local object. It&rsquo;s intended that </span><tt>method</tt><span> returns an integer and if that integer is above zero, then the checkbox is checked. Additional options on the input tag can be passed as a hash with </span><tt>options</tt><span>. The </span><tt>checked_value</tt><span>defaults to 1 while the default </span><tt>unchecked_value</tt><span> is set to 0 which is convenient for boolean values.</span></p>"
irb(main):017:0> sample_text.truncate_words(10)
=> "<p><span>Returns a checkbox tag tailored for accessing a specified attribute..."
irb(main):018:0>
irb(main):008:0> sample
```

**Comment by mackwic** [comment]
Can confirm, the bug is fixed on 4.2.6 with ruby 2.3.1
